frontend/app/routes/$lang+/_protected+/alerts+/subscribe+/confirm.tsx

```diff
@@ -166,28 +171,31 @@ export default function ConfirmSubscription() {
       <fetcher.Form className="max-w-prose" method="post" noValidate>
         <input type="hidden" name="_csrf" value={csrfToken} />
         <div className="mb-8 space-y-6">
-          <ContextualAlert type="info">
-            {!newCodeRequested ? (
-              <>
-                <p id="confirmation-information" className="mb-4">
-                  <Trans ns={handle.i18nNamespaces} i18nKey="alerts:confirm.confirmation-information-text" values={{ userEmailAddress: alertSubscription?.email }} />
-                </p>
-                <p id="confirmation-completed" className="mb-4">
-                  {t('alerts:confirm.confirmation-completed-text')}
-                </p>
-                <div className="grid gap-12 md:grid-cols-2">
-                  <p id="confirmation-language" className="mb-4 font-bold">
-                    {t('alerts:confirm.confirmation-selected-language')}
+          {!undefinedEmail && (
```

Comment:
I don't think you need to define a new variable for this. Wouldn't this work?

``` tsx
{ alertSubscription?.email && (
  <p>🦄</p>
)}
```

Reply:
`alertSubscription` should not be returned as `underfined` from the loader. If alertSubscription is undefined then the page should redirect or return a 404.

Reply:
I do kind of agree with @sebastien-comeau here.. I didn't think about the larger context.

Under what situation could this email become undefined, and does it make sense to render anything at all in that situation?

Reply:
Added the block of code to throw a 404.